### ISSUE TITLE
'notification' is a reserved word and can not be used as a key in FCM data payloads 

### ISSUE THREAD

**Issue by halaei** [kreait/firebase-php#588]
A validation is added by https://github.com/kreait/firebase-php/commit/d7fca18a3639d8898dcb6385ecf6d97df349d237 to disallow reserved keys. However, before this commit the 'notification' key was accepted by FCM and messages was delivered. So it is a breaking change for our app and validation may not be required. I try to use some other keyword for my next releases of Android apps. But maybe the commit should be reverted.

**Comment by halaei** [comment]
Maybe the firebase doc is incorrect, or maybe they only use the keyworkd in some clients we don't use. We currently only send FCM messages to our Android app.
Is it OK to give us option to opt out of the validation?

```php
    public function withData($data, $validate = true): self
    {
        $new = clone $this;
        $new->data = $data instanceof MessageData ? $data : MessageData::fromArray($data, $validate);

        return $new;
    }
    public static function fromArray(array $data, $validate = true): self
    {
       ...
            if ($validate) {
                self::assertValidKey((string) $key);
            }
    }
```

**Comment by jeromegamez** [comment]
Thanks for bringing this up! I just checked and I can confirm that the `notification` field is accepted although the official docs say they're not 😭

I'll provide a fix shortly!

**Comment by jeromegamez** [comment]
Here's the fix (https://github.com/kreait/firebase-php/commit/16fdb125f708f6e8cb20e6baf700c332e7123818), I will create a new release later today 🤞, thanks again!

**Comment by halaei** [comment]
Thanks

**Comment by jeromegamez** [comment]
Released with [5.17.1](https://github.com/kreait/firebase-php/releases/tag/5.17.1) 🚀
